state/relationunit.go

```diff
@@ -450,14 +450,24 @@ func (op *LeaveScopeOperation) internalLeaveScope() ([]txn.Op, error) {
 		Assert: txn.DocExists,
 		Remove: true,
 	}}
+
+	// Cross model relations might get out of sync between models.
+	// If this is not the first attempt to run the leave scope txns,
+	// force the removal of the relation rather than trying to do it cleanly.
+	_, isCrossModel, err := op.ru.relation.RemoteApplication()
+	if op.FatalError(errors.Trace(err)) {
+		return nil, err
+	}
+	forceRemove := isCrossModel && attempt > 0 && op.Force
```

Comment:
what is the side effect of dirty removal?

isCrossModel && op.Force might be sufficient here?

Reply:
So we hit this if the offering side has gone away and the consuming side gets out of sync. We still try to construct a reasonable txn slice the first time, but get out the bug hammer next time around to force model cleanup if needed.

Reply:
I've managed to revert it since other processing for Dead remote apps removes the need for this extra logic